### DISCUSSION TITLE
VirtualBox: Shared folders are sometimes not mounted in the PhotoPrism container

### DISCUSSION THREAD

**Discussion by 1oh1** [photoprism/photoprism#1709] [Q&A] [answered]
**What does not work as expected?**

*Background*: I have PhotoPrism running inside a Ubuntu 21.10 (5.13.0-20-generic) guest VM on a Windows 11 host over VirtualBox. I use VirtualBox's "shared folders" feature to auto-mount directories containing my images to the Ubuntu guest.

*Issue*: Sometimes, when I try to download images from my PhotoPrism instance, the download fails and I get an empty file. The failure is noticed only when trying to download full resolution versions of images and not when browsing through PhotoPrism generated thumbnails of various sizes. Looking at the logs, I see that the relevant full-res files are missing in the PhotoPrism container but I can open the missing files on the Ubuntu guest using a file manager. The path where the images are supposed to be is a VirtualBox "shared folder" that should've been mounted as a volume in the PhotoPrism docker container but doesn't seem to be.

**How can we reproduce it?**

1. Use VirtualBox (v6.1.28) to start a Ubuntu VM
2. Use [docker-compose.yml](https://github.com/photoprism/photoprism/blob/develop/docker/examples/docker-compose.yml) to start a PhotoPrism container with `docker-compose up`
3. Ensure VirtualBox's "shared folders" are mounted via the `volumes` section
```
volumes:
      ## The *originals* folder contains your original photo and video files (- "[host folder]:/photoprism/originals"):
      # All sf_XXXX are VirtualBox's 'shared folders'
      - "/media/sf_PN50-host:/photoprism/originals/PN50-host"
      - "/media/sf_PhotoPrism-Folders:/photoprism/originals/PhotoPrism-Folders"
```
4. Shut down and restart Ubuntu VM
5. Try to download an image from PhotoPrism web UI and see this error in Library > Errors:
`download: file “PhotoPrism-Folders/Himachal Trip (Nov 2021)/IMG_2021_12345.jpg” is missing`
6. Confirm this by running:
```
$ docker-compose exec photoprism bash
cd /photoprism/originals/PN50-host
cd <subfolder>
ls -al
total 8
drwxr-xr-x 2 root root 4096 Aug 19 21:30 .
drwxr-x--- 3 root root 4096 Aug 19 21:30 ..

cd ../../PhotoPrism-Folders/
total 8
drwxr-x--- 2 root       root       4096 Nov  7 21:10 .
drwxrwxrwx 1 photoprism photoprism 4096 Nov  8 17:29 ..
```

**What behavior do you expect?**
VirtualBox's "shared folders" are mounted properly to the PhotoPrism container after the guest VM is shutdown and rebooted

**Can you provide us with example files for testing or screenshots?**
[docker-compose.yml](https://pastebin.com/aY6SGSpB)

**What version you are using?**
PhotoPrism®
211106-cbfb3d48-Linux-x86_64

**Any other helpful information?**

The fix is pretty straightforward: `docker-compose down && docker-compose up -d`

However, having to do this every time I encounter this issue is a problem and I also noticed that if an image fails to download, it is automatically removed from PhotoPrism and I can't see it in the UI anymore. To get it back, I have to fix the volume mounting issue first and trigger a re-index.

I'm guessing this is a VirtualBox (or Windows) issue and not a PhotoPrism issue but I'm not sure.

**Comment by lastzero** [comment, marked as answer]
Yes, it's a VirtualBox issue. Sorry we can't really help with this!
